frontend/ui/widget/textboxwidget.lua

```diff
@@ -151,6 +152,26 @@ function TextBoxWidget:init()
         self.text_height = self.lines_per_page * self.line_height_px
     end
 
+    self:_computeDimensions()
+    self:_updateLayout()
+    if self.editable then
+        self:moveCursorToCharPos(self.charpos or 1)
+    end
+    self.dimen = Geom:new(self:getSize())
```

Comment:
I'm a bit unconfortable with these kind of changes. It might be all right, but after a few minutes looking around, I can't say I'm 100% sure - it's less readable that how it was before (_computeDimensions actually renders the text, so this naming is hiding abit the whole core of what's really happening).
And what about this self.dimen here that won't get updated by your setText() if you give it larger text that makes a few more lines ?

I'd be more confortable if you just did in your setText() : free() and then :init(). That's what we do in `InputText:initTextBox()` - it would feel less risky, and the wee additional recomputing this would do vs. your splitting would be bearable for the rare users of this new setText().

Reply:
`self.dimen` is a constructor-time only field, so it's irrelevant here ;).

Reply:
But I *like* catching design/implementation bugs with stuff like that ;o).